File: backend/app/schemas/cliente.py
import uuid
from datetime import datetime

from pydantic import BaseModel, EmailStr, field_validator

from app.models.cliente import TipoIdentificacion


class ClienteBase(BaseModel):
    tipo_identificacion: TipoIdentificacion
    numero_identificacion: str
    nombre: str
    apellido1: str | None = None
    apellido2: str | None = None
    razon_social: str | None = None
    email: EmailStr | None = None
    telefono: str | None = None
    provincia: str | None = None
    canton: str | None = None
    distrito: str | None = None
    direccion_exacta: str | None = None

    @field_validator("telefono")
    @classmethod
    def validate_telefono(cls, v: str | None) -> str | None:
        if v is not None:
            import re

            if not re.match(r"^[2-8]\d{7}$", v):
                raise ValueError("Teléfono debe ser 8 dígitos comenzando con 2-8")
        return v


class ClienteCreate(ClienteBase):
    pass


class ClienteUpdate(BaseModel):
    nombre: str | None = None
    apellido1: str | None = None
    apellido2: str | None = None
    razon_social: str | None = None
    email: EmailStr | None = None
    telefono: str | None = None
    provincia: str | None = None
    canton: str | None = None
    distrito: str | None = None
    direccion_exacta: str | None = None
    is_active: bool | None = None

    @field_validator("telefono")
    @classmethod
    def validate_telefono(cls, v: str | None) -> str | None:
        if v is not None:
            import re

            if not re.match(r"^[2-8]\d{7}$", v):
                raise ValueError("Teléfono debe ser 8 dígitos comenzando con 2-8")
        return v


class ClienteResponse(ClienteBase):
    id: uuid.UUID
    is_active: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/cliente.ts
export type TipoIdentificacion = "cedula_fisica" | "cedula_juridica" | "dimex" | "nite";

export interface Cliente {
  id: string;
  tipo_identificacion: TipoIdentificacion;
  numero_identificacion: string;
  nombre: string;
  apellido1: string | null;
  apellido2: string | null;
  razon_social: string | null;
  email: string | null;
  telefono: string | null;
  provincia: string | null;
  canton: string | null;
  distrito: string | null;
  direccion_exacta: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface ClienteCreate {
  tipo_identificacion: TipoIdentificacion;
  numero_identificacion: string;
  nombre: string;
  apellido1?: string;
  apellido2?: string;
  razon_social?: string;
  email?: string;
  telefono?: string;
  provincia?: string;
  canton?: string;
  distrito?: string;
  direccion_exacta?: string;
}

export interface ClienteUpdate {
  nombre?: string;
  apellido1?: string;
  apellido2?: string;
  razon_social?: string;
  email?: string;
  telefono?: string;
  provincia?: string;
  canton?: string;
  distrito?: string;
  direccion_exacta?: string;
  is_active?: boolean;
}
